use event.key for search shortcut

gabalafou opened the pull request:
Fixes #1521

trallard commented on src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js:
```diff
@@ -236,12 +236,18 @@ var addEventListenerForSearchKeyboard = () => {
-      if ((event.ctrlKey || event.metaKey) && event.code == "KeyK") {
+      if (
+        (event.ctrlKey || event.metaKey) &&
+        (/k/i.test(event.key) || event.keyCode === 75)
```
Since this might seem obscure for folks unfamiliar with non-QWERTY layouts, perhaps a comment like this one would be good
```suggestion
		// Needed for DVORAK key-mapping
        (/k/i.test(event.key) || event.keyCode === 75)
```

12rambau replied:
why not simply "non en-iso" instead of "DVORAK" ? If I understand correctly this fix is for any keyboard mapping right ?

gabalafou replied:
I don't think it really needs commenting. Most keyboard shortcuts should match the **letter** (event.key) that would appear in a text editor or input field and **not** the physical **key** (event.code) on your keyboard. I know, I know, the terminology is confusing, but that's the way it is. 

The exception would be if you're really trying to match the shortcut to a specific physical place on the keyboard. For example, imagine if you were trying to create an app that turns your keyboard into a piano-like musical instrument. Then you would want to map the physical location of keys on the keyboard rather than the letters. 

I will add, however, that it's a little fuzzy to me what should happen when apps are internationalized. In a French computer, for example, I think `ctrl + t` opens a new browser tab, even though the word for tab in French, which is "onglet", does not begin with a "t". But it's weird to me that keyboard shortcuts that were chosen because of a first-letter mnemonic in English (for example: `ctrl+s` because **S**ave begins with the letter "s") should stay the same across languages.

But the bottom line in our case is that if we are showing the keyboard shortcut in the search bar across all languages as `ctrl + k` then we should bind to the letter, not the key.

Read [event.code and event.key](https://javascript.info/keyboard-events#event-code-and-event-key) for more explanation.

I think leaving the code without a comment will actually be less confusing than adding a somewhat cryptic comment about key-mapping because standard practice is to use `event.key` for most keyboard shortcuts.

trallard commented on src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js:
```diff
@@ -236,12 +236,18 @@ var addEventListenerForSearchKeyboard = () => {
-      else if (document.activeElement === input && event.code == "Escape") {
+      else if (
+        document.activeElement === input &&
+        (/escape/i.test(event.key) || event.keyCode === 27)
```
Same as above?

trallard reviewed (approved):
I had to remap my keyboard to test, and it seems to work 👍🏽 left some not blocking suggestions but otherwise I'd be happy to have this merged

drammock reviewed (commented):
a few questions:

1. `keyCode` [is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode). The example on that page uses `event.key` first and falls back to `event.keyCode` if `key` is undefined, which is not exactly what we're doing here (here if `event.key` is defined but doesn't match, you still check `event.keyCode`). The linked page has a long discussion of what different browsers/OSes return when asked for `keyCode` which makes me wonder if this is a can of worms that we don't want to open. Can we really not rely only on `event.key` here?

2. is a case-insensitive match really what we want? I can see arguments either way ("make it still work if caps lock is on" versus "don't hijack Ctrl+Shift+k which might have been mapped to something else by the browser/OS")

3. here we allow either Ctrl+k or Meta+k on any platform. I wonder if we should be more strict and triage the modifier based on platform (again, in the name of not hijacking other key combos that may have been mapped to other actions by the browser or OS)

gabalafou commented:
Thanks @drammock, I updated the PR based on your questions/suggestions. I will also reply to them one by one.

1. I've spent too much time in Jupyter land 😅. JupyterLab code was originally written to support legacy browsers, so you'll find code snippets that use `keyCode`. But I imagine we're all fine with this [compatibility table ](https://caniuse.com/?search=event.key), so I switched the code to just use `event.key`.
2. Good catch. I think it should not work with shift, but it should work with caps lock. I modified the code accordingly.
3. I think your hunch is right and we should protect against stealing other shortcuts. So I've made the shortcut more strict. (Now it should be a 1-to-1 match with what we show the user in the search box. In other words, only when `isMac` is true do we show `⌘` + `K` in the search box; otherwise we show `Ctrl` + `K`. The shortcut handler now follows the exact same logic.)
